File: LocalPatterns/app/src/main/java/com/sunday/study/one/leaks/aops/login/LoginActivity.java
package com.sunday.study.one.leaks.aops.login;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

import com.sunday.study.one.R;
import com.sunday.study.one.leaks.App;

public class LoginActivity extends AppCompatActivity {


    //Thread Local
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
           setContentView(R.layout.activity_loginaop);
    }

    @CheckLogin
    public void logins(View login){
        Log.e("stags","stars");
    }

    //So a junk foods
    public void check(View login){
        Log.e("stags","starss");
        App.isLogin = !App.isLogin;

    }

}

File: LocalPatterns/JetpackDemo/src/main/java/com/example/sunday/jetpackdemo/databindings/LoginActivity.kt
package com.example.sunday.jetpackdemo.databindings

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.sunday.jetpackdemo.R
import com.example.sunday.jetpackdemo.databinding.ActivityLoginBinding

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

         var mBinding : ActivityLoginBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
//         var model = LoginModel()
          mBinding.model = LoginModel()
    }
}